pkg/istio/client.go

```diff
@@ -382,3 +392,59 @@ func prepareIstioHTTPRouteForCR(obj *v1alpha1.Watcher) *istioapi.HTTPRoute {
 func destinationHost(serviceName, serviceNamespace string) string {
 	return fmt.Sprintf("%s.%s.svc.cluster.local", serviceName, serviceNamespace)
 }
+
+// SyncCertificateSecretToIstio fetches the CA Root Certificate, configured by KLM flags, and creates, updates
+// or deletes a copy of it in the Istio namespace. This is needed since Istio needs the Root CA Certificate secret
+// in its own namespace to use it for Gateways with strict mTLS connection.
+func (c *Client) SyncCertificateSecretToIstio(ctx context.Context, kcpClient client.Client) error {
+	certSecret := &corev1.Secret{}
+	err := kcpClient.Get(ctx,
+		client.ObjectKey{
+			Namespace: c.config.WatcherRootCertificateNamespace,
+			Name:      c.config.WatcherRootCertificateName,
+		}, certSecret)
+
+	if apierrors.IsNotFound(err) {
+		// if CA Certificate does not exist, check if it exists in istio Namespace, if yes remove it
```

Comment:
Could you explain to me again why we don't directly sync it into the correct namespace always?